features/steps/givens/attributes.py

```diff
@@ -71,6 +73,19 @@ def step_impl(context, attribute):
     setattr(context, 'attribute', attribute)
 
 
+@validate_step('Its final segment')
+def step_impl(context):
+    context._push()
+    context.instances = list()
+    for curves in context._stack[1]["instances"]:
```

Comment:
Why context._stack[1]["instances'], doesn't simply `context.instances` work?
I just tried in my debugger, and it indeed doesn't. I'd expect them to be the same. 
What kind of errors did you encounter here?

Reply:
I didn't encounter any errors.  This was the first way I put it together and it worked, so I moved on :smiley:.  Note L78 where context is pushed, meaning `context.instances` then becomes `None`, then turned into an empy list on L79..